src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/ProtobufSerializerHelper.cs

```diff
@@ -0,0 +1,233 @@
+            index++;
+        }
+
+        // If fewer than 3 bytes were used, pad with zeros
+        while (index < 2)
+        {
+            buffer[currentPosition++] = 0x80;
+            index++;
+        }
+
+        while (index < 3)
+        {
+            buffer[currentPosition++] = 0x00;
+            index++;
+        }
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void WriteRawVarint32(byte[] buffer, ref int currentPosition, uint value)
+    {
+        // Optimize for the common case of a single byte value
+        if (value < 128 && currentPosition < buffer.Length)
+        {
+            buffer[currentPosition++] = (byte)value;
+            return;
+        }
+
+        // Fast path when capacity is available
+        while (currentPosition < buffer.Length)
+        {
+            if (value > 127)
+            {
+                buffer[currentPosition++] = (byte)((value & 0x7F) | 0x80);
+                value >>= 7;
+            }
+            else
+            {
+                buffer[currentPosition++] = (byte)value;
+                return;
+            }
+        }
+
+        // Write byte individually
+        // We dont refresh the buffer but this could be used when the buffer is refreshed.
+        // Right now, it would simply fail.
+        while (value > 127)
+        {
+            WriteRawByte(buffer, ref currentPosition, (byte)((value & 0x7F) | 0x80));
+            value >>= 7;
+        }
+
+        WriteRawByte(buffer, ref currentPosition, (byte)value);
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void WriteRawVarint64(byte[] buffer, ref int currentPosition, ulong value)
+    {
+        // Optimize for the common case of a single byte value
+        if (value < 128 && currentPosition < buffer.Length)
+        {
+            buffer[currentPosition++] = (byte)value;
+            return;
+        }
+
+        // Fast path when capacity is available
+        while (currentPosition < buffer.Length)
+        {
+            if (value > 127)
+            {
+                buffer[currentPosition++] = (byte)((value & 0x7F) | 0x80);
+                value >>= 7;
+            }
+            else
+            {
+                buffer[currentPosition++] = (byte)value;
+                return;
+            }
+        }
+
+        // Write byte individually
+        // We dont refresh the buffer but this could be used when the buffer is refreshed.
+        // Right now it would simply fail.
+        while (value > 127)
+        {
+            WriteRawByte(buffer, ref currentPosition, (byte)((value & 0x7F) | 0x80));
+            value >>= 7;
+        }
+
+        WriteRawByte(buffer, ref currentPosition, (byte)value);
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void WriteRawByte(byte[] buffer, ref int currentPosition, byte value)
+    {
+        if (currentPosition < 0)
```

Comment:
When we are incrementing the currentPosition, how would be end up with value less than zero.

Reply:
good catch. Fixed.